test/e2e/cypress/pageObject/base_po.js

```diff
@@ -71,13 +71,25 @@ export const clickUserDropdownProfileButton = () => {
   return cy.get(userDropdownProfileButton).click();
 };
 
-export const typeTotpCode = (totpSecret, inputField) => {
-  const { otp } = TOTP.generate(totpSecret);
-  return cy
-    .get(inputField)
-    .clear()
-    .type(otp)
-    .then(() => totpSecret);
+export const typeNextGeneratedTotpCode = (
+  totpSecret,
+  inputField,
+  forceNext = false
+) => {
+  const currentTime = Date.now();
+  const totpDuration = 30000;
+  const expirationTime = Math.ceil(currentTime / totpDuration) * totpDuration;
+  const remainingTime = Math.floor(expirationTime - currentTime);
+  const timeToWait =
+    forceNext === false && remainingTime > 10000 ? 0 : remainingTime;
+  return cy.wait(timeToWait).then(() => {
```

Comment:
i would not mind an empty space between const de0clarations and the return statement, just for the readability